### ISSUE TITLE
Add option to rule semi to omit last statement semicolon

### ISSUE THREAD

**Issue by bbuhler** [eslint/eslint#9524]
**What rule do you want to change?**
semi

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

```js
export default {
  foo: 'bar',
  doStuff() {
    return function() {
      const foo = 'bar';
      return foo + ' yäy';
    }
  }
}
```

**What does the rule currently do for this code?**
```
test.js
     8:6   error  Missing semicolon  semi
    10:2   error  Missing semicolon  semi
```

**What will the rule do after it's changed?**
With the new option set to true it will not warn for missing semicolon of the last statement in a block scope.

**Comment by bbuhler** [comment]
Okay I missed the part behind the brackets in the documentation.

> ignores the last semicolon in a block in which its braces (and therefore the content of the block) *are in the same line*

But is there a way to achieve my expected behavior?

**Comment by platinumazure** [comment]
@bbuhler Taking a look at the docs, I don't think there currently is a way to do this.

Maybe we could consider a proposal of adding a new option, maybe something like `omitIfLastStatementInBlock` (or something less verbose).

Could you please amend your original issue post to follow our [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)? Thanks!

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
